File: Java/Easy/Determine-if-String-Halves-Are-Alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int n = s.length();
    return countVowels(s.substring(0,n/2)) == countVowels(s.substring(n/2));
    }

    public int countVowels(String s) {
        int count = 0;
        for (int i = 0; i<s.length(); i++) {
            switch (s.charAt(i)) {
                case 'a':
                case 'e':
                case 'i':
                case 'o':
                case 'u':
                case 'A':
                case 'E':
                case 'I':
                case 'O':
                case 'U':
                    count ++;
            }
                
        }
        return count;
    }
}

File: Python/Easy/Determine-if-String-Halves-Are-Alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = set('aeiouAEIOU')
        n = len(s) // 2
        numLeft = sum(1 for c in s[:n] if c in vowels)
        numRight = sum(1 for c in s[n:] if c in vowels)
        return numLeft == numRight
            
